src/System.Net.Requests/src/System/Net/FileWebRequest.cs

```diff
@@ -0,0 +1,453 @@
+                {
+                    lock (thisRef)
+                    {
+                        if (thisRef._writePending || thisRef._writing)
+                        {
+                            thisRef._blockReaderUntilRequestStreamDisposed = new ManualResetEventSlim();
+                        }
+                    }
+                }
+                thisRef._blockReaderUntilRequestStreamDisposed?.Wait();
+
+                try
+                {
+                    return thisRef._response ?? (thisRef._response = new FileWebResponse(thisRef, thisRef._uri, thisRef._fileAccess, !thisRef._syncHint));
+                }
+                catch (Exception e)
+                {
+                    throw new WebException(e.Message, e);
+                }
+            }, this, CancellationToken.None, TaskCreationOptions.DenyChildAttach, TaskScheduler.Default), callback, state);
+        }
+
+        public override Stream EndGetRequestStream(IAsyncResult asyncResult)
+        {
+            Stream stream = TaskToApm.End<Stream>(asyncResult);
+            _writePending = false;
+            return stream;
+        }
+
+        public override WebResponse EndGetResponse(IAsyncResult asyncResult)
+        {
+            WebResponse response = TaskToApm.End<WebResponse>(asyncResult);
+            _readPending = false;
+            return response;
+        }
+
+        public override Stream GetRequestStream()
+        {
+            IAsyncResult result = BeginGetRequestStream(null, null);
+
+            if (Timeout != Threading.Timeout.Infinite &&
+                !result.IsCompleted &&
+                (!result.AsyncWaitHandle.WaitOne(Timeout, false) || !result.IsCompleted))
+            {
+                _stream?.Close();
+                throw new WebException(SR.net_webstatus_Timeout, WebExceptionStatus.Timeout);
+            }
+
+            return EndGetRequestStream(result);
+        }
+
+        public override WebResponse GetResponse()
+        {
+            _syncHint = true;
+            IAsyncResult result = BeginGetResponse(null, null);
+
+            if (Timeout != Threading.Timeout.Infinite &&
+                !result.IsCompleted &&
+                (!result.AsyncWaitHandle.WaitOne(Timeout, false) || !result.IsCompleted))
+            {
+                _response?.Close();
+                throw new WebException(SR.net_webstatus_Timeout, WebExceptionStatus.Timeout);
+            }
+
+            return EndGetResponse(result);
+        }
+
+        internal void UnblockReader()
+        {
+            lock (this) { _blockReaderUntilRequestStreamDisposed?.Set(); }
+            _writing = false;
+        }
+
+        public override bool UseDefaultCredentials
+        {
+            get { throw NotImplemented.ByDesignWithMessage(SR.net_PropertyNotImplementedException); }
+            set { throw NotImplemented.ByDesignWithMessage(SR.net_PropertyNotImplementedException); }
```

Comment:
Is this the behavior from Desktop?  It's rare we would throw an exception on a property getter.